mlir/lib/Dialect/LLVMIR/IR/LLVMInterfaces.cpp

```diff
@@ -58,7 +58,15 @@ mlir::LLVM::detail::verifyAliasAnalysisOpInterface(Operation *op) {
   ArrayAttr tags = iface.getTBAATagsOrNull();
   if (!tags)
     return success();
-
+  if (tags.size() > 0) {
```

Comment:
Shouldn't it just be:
`return isArrayOf<TBAATagAttr>(op, tags) || isArrayOf<TBAAAccessTagAttr>(op, tags);`

Reply:
The `isArrayOf<xxx>` reports an error if any element in the provided array is not type of xxx.
and the result type `LogicalResult` dose not support the `||` operator.

```
/// Verifies that all elements of `array` are instances of `Attr`.
template <class AttrT>
static LogicalResult isArrayOf(Operation *op, ArrayAttr array) {
  for (Attribute iter : array)
    if (!isa<AttrT>(iter))
      return op->emitOpError("expected op to return array of ")
             << AttrT::getMnemonic() << " attributes";
  return success();
}
```
see 
https://github.com/llvm/llvm-project/blob/main/mlir/lib/Dialect/LLVMIR/IR/LLVMInterfaces.cpp#L19-L27

Reply:
Ok I understand. I would suggest to extend the AliasAnalysisOpInterface. It currently has one function that returns the TBAA tags. How about implementing two functions:
```
iface.getTBBATagsOrNull()
```
and
```
iface.getTBBAAccessTagsOrNull()
```
This could potentially be implemented by having two arrays stored on the alias operations. Or by checking if the array contains the old or the new kind of TBAA attributes. That way the verifier can use the existing logic for the two cases separately. And to users of the exiting TBAA attributes don't need to understand the new kind of TBAA metadata.

WDYT?